src/ws_server.rs

```diff
@@ -0,0 +1,67 @@
+use std::thread;
+use context::{ ContextTrait, SharedContext };
+use ws::{Handler, Sender, Result, Message, Handshake, CloseCode, Error};
+
+use ws::listen;
+
+pub struct WsServer;
+
+pub struct WsHandler {
+    pub out: Sender,
+    pub context: SharedContext
+}
+
+impl WsServer {
+
+    pub fn start(context: SharedContext) {
+        thread::Builder::new().name("WsServer".to_owned()).spawn(move || {
+
+            let hostname;
+            let port;
+            {
+                let ctx = context.lock().unwrap();
+                hostname = ctx.hostname.clone();
+                port = ctx.ws_port;
+            }
+
+            let host_port = (&hostname as &str, port);
+
+            listen(host_port, |out| {
+                WsHandler {
+                    out: out,
+                    context: context.clone(),
+                }
+            }).unwrap();
+        }).unwrap();
+    }
+}
+
+impl Handler for WsHandler {
+
+    fn on_open(&mut self, _: Handshake) -> Result<()> {
+        println!("Hello new ws connection");
+
+        self.context.lock().unwrap().add_socket(self.out.clone());
+
+        Ok(())
+    }
+
+    fn on_message(&mut self, msg: Message) -> Result<()> {
+        // echo the message back to the client
+        self.out.send(msg)
```

Comment:
Not sure if this is a good idea to send back the message to the client. Let's just log for now.